ssl/statem/statem_srvr.c

```diff
@@ -1400,6 +1400,8 @@ CON_FUNC_RETURN dtls_construct_hello_verify_request(SSL_CONNECTION *s,
         return CON_FUNC_ERROR;
     }
 
+    s->d1->hello_verify_request = SSL_HVR_SENT;
```

Comment:
Perhaps here also mention that is in support of the *stateful* DTLS listener case, where version selection happens during receipt of the original hello, but we need to save away the downgrade state until the HVR response is processed.